server/repository/reactionRepository.js

```diff
@@ -0,0 +1,91 @@
+import Model from '../models';
+import notificationRepo from './notificationRepository';
+
+const { Reaction } = Model;
+
+/**
+ * Article repository class
+ */
+class ReactionRepository {
+  /**
+     * Function to like an article in the database
+     * @param {integer} articleId
+     * @param { integer } userId
+     * @param { integer } reactionType
+     * @returns {integer} article
+     ** otherwise it throws an error
+     */
+  static async createReaction(articleId, userId, reactionType) {
+    const reaction = await Reaction.create({
+      reactionType,
+      articleId,
+      userId
+    });
+    notificationRepo.createNotification({
```

Comment:
@ProfJigsaw Usually the idea of a repository is to have proper separation of concerns between the models. I don't think we should call the notification repo here.

Reply:
@PriscillaSam Yes, I thought of that but if I remove this one-time call from the repo then I have to sacrifice DRY in the controllers. This is because there are several cases on creating reactions and comments and each one would need a call to createNotification. This would be more bogus.

Reply:
Fixed. Thanks for the catch